### PR TITLE
Add roomsettings for tournaments

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

File: server/tournaments/index.ts [hunk 1]
```diff
@@ -1793,18 +1836,270 @@ const commands: ChatCommands = {
...
+					room.settings.tournaments.playerCap = num;
+					if (tour && !tour.isTournamentStarted && !tour.playerCap) {
+						this.parse(`/tour setplayercap ${num}`);
+						if (room.settings.tournaments.autostart === true) this.parse(`/tour autostart on`);
+					}
+					room.saveSettings();
+					this.privateModAction(`Player Cap was set to ${num} for every tournament by ${user.name}`);
+					this.modlog('TOUR SETTINGS', null, `playercap: ${num}`);
+					if (Config.tourdefaultplayercap && room.settings.tournaments.playerCap > Config.tourdefaultplayercap) {
+						Monitor.log(`[TourMonitor] Room ${room.roomid} setting cap for every tour over default cap (${room.settings.tournaments.playerCap})`);
```
Comment (mia-pi-git) [inline server/tournaments/index.ts:2082]: Why is this monitored somewhere they won't see it? Why not just errorReply?

File: server/tournaments/index.ts [hunk 2]
```diff
@@ -1831,15 +2126,66 @@ const commands: ChatCommands = {
...
 	createTournament,
 	commands,
+	roomSettings,
 };
+
+for (const room of Rooms.rooms.values()) {
+	// @ts-ignore
+	const announcements = room.settings.tourAnnouncements;
+	// @ts-ignore
+	delete room.settings.tourAnnouncements;
```
Comment (Zarel) [inline server/tournaments/index.ts:2183]:
```suggestion
	const announcements = (room.settings as any).tourAnnouncements;
	delete (room.settings as any).tourAnnouncements;
```

I'd prefer `any`-casting to `ts-ignore`, going forward.

(The reason is that `any`-casting only suppresses some errors, while `ts-ignore` suppresses every error, maybe including some actual bugs.)

**Review by Zarel — APPROVED** [review]
Looks good to me! Other than the `ts-ignore` thing.

**Comment by KrisXV** [comment]
done @Zarel

**Pull request merged**